products/caasp/main.pm

```diff
@@ -167,7 +167,9 @@ else {
 }
 
 # ==== Extra tests run after installation  ====
-if (get_var('REGISTER')) {
+# REGISTER = 'suseconnect' -> Registers with SCC after the installation
+# REGISTER = 'installation' -> Registers with SCC during the installation
```

Comment:
Now I think your comments are helpful but adding this comments here seems like not the right place. How about extending the file "variables" with a check for each used variable if it's properly accounted for?

Reply:
The reason I've put the comments into this place is actually because of the [comment before](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/6c7304e957f040bbbd236881bede59ac42ffbf37/products/caasp/main.pm#L169) is kind of misleading. Depending on the value of the `REGISTER` variable, the system either registers *after* or *during* the installation. When I originally read the source code, I though that the purpose of checking the `REGISTER` variable is to trigger (or not) the [register_and_check](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/8e7a2a8e5c3892a8f23d8c455d135b00b0d3fdba/tests/caasp/register_and_check.pm) which always takes place always *after* the installation. So the comment was correct, but I didn't know that there are *two different tests for the same thing* (registration). After [reading this](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/8e7a2a8e5c3892a8f23d8c455d135b00b0d3fdba/tests/caasp/register_and_check.pm#L11), then [I found out](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/6c7304e957f040bbbd236881bede59ac42ffbf37/products/caasp/main.pm#L76-L77) that the registration is possible to take place also during the installation of the system. So, I intentionally put this comment on that spot, simply because it was the one introduced my confusion in the first place.

I agree that the best place is not this one, but at the that moment I wasn't aware of any better. For sure, this "variables" file sounds like a better approach -- I didn't know about it (happy to learn new things every day :D ). Personally, in respect of CaaSP, I think we should design a UML chart. The test is already complicated with mutex and stuff and it's going to be even more complicated in the future.

But anyway, apart from the comments, the purpose of this PR makes sure that CaaSP registration is protected from user error (wrong configuration of the controller in openQA). So if this the only *blocker*, I am happy to relocate/remove the comments to another place :)

Reply:
it's fine for this PR, would be appreciated if you take a look to improve later